config/label.go

```diff
@@ -89,3 +110,30 @@ func parseIntLabel(labels map[string]string, key string) (intVal int, err error)
 	}
 	return
 }
+
+func (l *LabelConfig) getImageFromKubeDeployment(appName string) (imageName string, err error) {
+	clientset, err := utils.NewKubeClientset(l.KubeConfig, l.KubeNamespace)
+	if err != nil {
+		log.Errorf("Error creating Kubernetes clientset: %v", err)
+		return
+	}
+
+	d, err := clientset.Extensions().Deployments(l.KubeNamespace).Get(appName)
+	if err != nil {
+		log.Errorf("Error getting deployment %s: %v", appName, err)
+		return
+	}
+
+	podSpec := d.Spec.Template.Spec
+	containers := len(podSpec.Containers)
+
+	if containers == 1 {
+		imageName = podSpec.Containers[0].Image
+		log.Debugf("Got image %s for deployment %s", imageName, appName)
+	} else {
+		// TODO!! Support pods with multiple containers
```

Comment:
Yes in K8s a container always belongs to a pod and that is what is scaled. Single container pods are very common (all the MicroBadger pods are single container). To support multi container pods we could make the container name part of the apps JSON.

I think there is a gap in container metadata because Docker labels are not exposed as Kubernetes annotations.